<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Подготовка" data-toc-modified-id="Подготовка-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Подготовка</a></span></li><li><span><a href="#Обучение" data-toc-modified-id="Обучение-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Обучение</a></span></li><li><span><a href="#Выводы" data-toc-modified-id="Выводы-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Выводы</a></span></li></ul></div>

# Проект для «Викишоп»

Интернет-магазин «Викишоп» запускает новый сервис. Теперь пользователи могут редактировать и дополнять описания товаров, как в вики-сообществах. То есть клиенты предлагают свои правки и комментируют изменения других. Магазину нужен инструмент, который будет искать токсичные комментарии и отправлять их на модерацию. 

Обучу модель классифицировать комментарии на позитивные и негативные. В моем распоряжении набор данных с разметкой о токсичности правок.



**Задачи**

1. Загрузить и подготовить данные.
2. Обучить разные модели. 
3. Выбрать лучшую модель.



## Подготовка

In [1]:
!pip install pymystem3
!pip3 install catboost
!pip install lightgbm
!pip install scikit-learn==1.1.3

In [38]:
import pandas as pd
from pymystem3 import Mystem
import nltk
from nltk.stem import WordNetLemmatizer
from nltk.corpus import stopwords
from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer, TfidfTransformer
from sklearn.model_selection import train_test_split, GridSearchCV, RandomizedSearchCV
from sklearn.metrics import f1_score
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.pipeline import Pipeline
import numpy as np
from catboost import CatBoostClassifier
from sklearn.metrics import  make_scorer
import lightgbm as lgb
from sklearn.dummy import DummyClassifier
import re 
from scipy.sparse import vstack



In [2]:
nltk.download('wordnet')
nltk.download('stopwords')
nltk.download('punkt')
nltk.download('averaged_perceptron_tagger')
nltk.download('omw-1.4')

[nltk_data] Downloading package wordnet to
[nltk_data]     C:\Users\iyaro\AppData\Roaming\nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\iyaro\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\iyaro\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     C:\Users\iyaro\AppData\Roaming\nltk_data...
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!
[nltk_data] Downloading package omw-1.4 to
[nltk_data]     C:\Users\iyaro\AppData\Roaming\nltk_data...
[nltk_data]   Package omw-1.4 is already up-to-date!


True

In [3]:
RS=12345

In [4]:
try:
    data = pd.read_csv('/datasets/toxic_comments.csv', index_col='Unnamed: 0')
except:
    data = pd.read_csv(r'C:\Users\iyaro\Downloads\toxic_comments.csv', index_col='Unnamed: 0')

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 159292 entries, 0 to 159450
Data columns (total 2 columns):
 #   Column  Non-Null Count   Dtype 
---  ------  --------------   ----- 
 0   text    159292 non-null  object
 1   toxic   159292 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 3.6+ MB


In [6]:
data['text'] = data['text'].apply(str)


In [7]:
data.head(5)

text  toxic
0  Explanation\nWhy the edits made under my usern...      0
1  D'aww! He matches this background colour I'm s...      0
2  Hey man, I'm really not trying to edit war. It...      0
3  "\nMore\nI can't make any real suggestions on ...      0
4  You, sir, are my hero. Any chance you remember...      0

In [8]:
stop_words = set(stopwords.words('english'))

In [9]:
%%time
lemmatizer = WordNetLemmatizer()
regex_alpha = re.compile('[^a-zA-Z]+')
def lemma(text):
    token_pos_tuples = nltk.pos_tag([
        token.lower() for token in re.sub(regex_alpha, ' ', text).split()
        if token not in stop_words
    ])
    lemmas = [
        lemmatizer.lemmatize(token, pos='n')
        if pos.startswith('N')
        else lemmatizer.lemmatize(token, pos='v')
        if pos.startswith('V')
        else lemmatizer.lemmatize(token, pos='a')
        if pos.startswith('J')
        else lemmatizer.lemmatize(token, pos='r')
        if pos.startswith('R')
        else lemmatizer.lemmatize(token)
        for token, pos in token_pos_tuples
    ]
    return ' '.join(lemmas)
data['text_lemmatized'] = data['text'].apply(lambda x: lemma(x))

Wall time: 4min 44s


In [10]:
data.head(20)

text  toxic  \
0   Explanation\nWhy the edits made under my usern...      0   
1   D'aww! He matches this background colour I'm s...      0   
2   Hey man, I'm really not trying to edit war. It...      0   
3   "\nMore\nI can't make any real suggestions on ...      0   
4   You, sir, are my hero. Any chance you remember...      0   
5   "\n\nCongratulations from me as well, use the ...      0   
6        COCKSUCKER BEFORE YOU PISS AROUND ON MY WORK      1   
7   Your vandalism to the Matt Shirvington article...      0   
8   Sorry if the word 'nonsense' was offensive to ...      0   
9   alignment on this subject and which are contra...      0   
10  "\nFair use rationale for Image:Wonju.jpg\n\nT...      0   
11  bbq \n\nbe a man and lets discuss it-maybe ove...      0   
12  Hey... what is it..\n@ | talk .\nWhat is it......      1   
13  Before you start throwing accusations and warn...      0   
14  Oh, and the girl above started her arguments w...      0   
15  "\n\nJuelz Santanas Age\n\nIn 2002, Juelz Sant...      0   
16  Bye! \n\nDon't look, come or think of comming ...      1   
17   REDIRECT Talk:Voydan Pop Georgiev- Chernodrinski      0   
18  The Mitsurugi point made no sense - why not ar...      0   
19  Don't mean to bother you \n\nI see that you're...      0   

                                      text_lemmatized  
0   explanation why edits make username hardcore m...  
1   d aww he match background colour i seemingly s...  
2   hey man i really try edit war it guy constantl...  
3   more i make real suggestion improvement i wond...  
4               you sir hero any chance remember page  
5              congratulation well use tool well talk  
6        cocksucker before you piss around on my work  
7   your vandalism matt shirvington article revert...  
8   sorry word nonsense offensive anyway i intend ...  
9                alignment subject contrary dulithgow  
10  fair use rationale image wonju jpg thanks uplo...  
11                    bbq man let discuss maybe phone  
12  hey talk what exclusive group wp taliban good ...  
13  before start throw accusation warning let revi...  
14  oh girl start argument she stick nose belong i...  
15  juelz santanas age in juelz santana year old c...  
16        bye don look come think comming back tosser  
17    redirect talk voydan pop georgiev chernodrinski  
18  the mitsurugi point make sense argue include h...  
19  don mean bother i see write something regard r...

In [11]:
x = data.drop('toxic', axis=1)
y = data['toxic']
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.5, stratify=y, random_state=RS)
x_test, x_valid, y_test, y_valid = train_test_split(x_test, y_test, test_size=0.5, stratify=y_test, random_state=RS)


In [12]:
vectorizer = TfidfVectorizer(stop_words=stop_words)
train_tfidf = vectorizer.fit_transform(x_train['text_lemmatized'])
valid_tfidf = vectorizer.transform(x_valid['text_lemmatized'])
test_tfidf = vectorizer.transform(x_test['text_lemmatized'])

# Вывод:
Данные загружены, текст лемматизирован, выборки поделены и применен TF-IDF. Можно перейти к обучению.

## Обучение

Обучу 4 модели. Catboost, lgb, случайный лес и логистическую регрессию.

In [34]:
x_train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 79646 entries, 106251 to 55812
Data columns (total 2 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   text             79646 non-null  object
 1   text_lemmatized  79646 non-null  object
dtypes: object(2)
memory usage: 1.8+ MB


In [42]:
%%time
pipeline = Pipeline([
    ('vect', TfidfVectorizer()),
    ('model', LogisticRegression(random_state=RS))
])

parameters = {
    'vect__max_df': (0.6, 0.8, 1.0),
    'vect__min_df': (0, 1, 2, 5),
    'vect__stop_words': ('english', list(stop_words)),
    'vect__ngram_range': ((1, 1), (1, 2)),
    'model__solver': ['lbfgs', 'newton-cg', 'liblinear'],
    'model__C' : np.arange(1, 11, 2),
    'model__warm_start': [True, False],
    'model__max_iter': range(100, 500),
    'model__class_weight': [None, 'balanced']
}

pipeline_rscv = RandomizedSearchCV(
    estimator=pipeline,
    param_distributions=parameters,
    n_iter=10,
    cv=5,
    random_state=RS,
    scoring='f1'
)
pipeline_rscv.fit(x_train['text_lemmatized'], y_train)

Wall time: 7min 12s


RandomizedSearchCV(cv=5,
                   estimator=Pipeline(steps=[('vect', TfidfVectorizer()),
                                             ('model',
                                              LogisticRegression(random_state=12345))]),
                   param_distributions={'model__C': array([1, 3, 5, 7, 9]),
                                        'model__class_weight': [None,
                                                                'balanced'],
                                        'model__max_iter': range(100, 500),
                                        'model__solver': ['lbfgs', 'newton-cg',
                                                          'liblinear'],
                                        'model__warm_start': [True, False],
                                        'vect__max_df':...0.8, 1.0),
                                        'vect__min_df': (0, 1, 2, 5),
                                        'vect__ngram_range': ((1, 1), (1, 2)),
                                        'vect__stop_words': ('english',
                                                             ['o', 'you', 'we',
                                                              'yourself', 'and',
                                                              'between', 'ma',
                                                              'because', 'it',
                                                              'had', 'shouldn',
                                                              'my', 'after',
                                                              'y', "don't",
                                                              'yours',
                                                              'against',
                                                              'herself', 'at',
                                                              'ours', 'her',
                                                              'has', 'while',
                                                              'their', 'any',
                                                              'that', "didn't",
                                                              'then', 'if',
                                                              'your', ...])},
                   random_state=12345, scoring='f1')

In [43]:
best_params = pipeline_rscv.best_params_
best_score = pipeline_rscv.best_score_

print("Лучшие параметры:", best_params)
print("Лучшая оценка:", best_score)

Лучшие параметры: {'vect__stop_words': 'english', 'vect__ngram_range': (1, 2), 'vect__min_df': 0, 'vect__max_df': 0.6, 'model__warm_start': False, 'model__solver': 'lbfgs', 'model__max_iter': 286, 'model__class_weight': 'balanced', 'model__C': 9}
Лучшая оценка: 0.7682465571259567


Логистическая регрессия не справилась с минимальным порогом в 0.75

In [14]:
f1_scorer = make_scorer(f1_score)

In [33]:
%%time
model_forest = RandomForestClassifier(random_state=RS)
params = {
    'model__n_estimators': [40, 70, 110],
    'model__max_depth': [1, 9, 2],
    'model__min_samples_split': [2, 5, 10]
}

pipeline = Pipeline([
    ('vect', CountVectorizer(min_df=10, max_df=1.0, ngram_range=(1, 4))),
    ('tfidf', TfidfTransformer()),
    ('model', RandomForestClassifier(random_state=RS))
])

grid = GridSearchCV(pipeline, cv=3, n_jobs=-1, param_grid=params, verbose=3, scoring=f1_scorer)
grid.fit(x_train['text_lemmatized'], y_train)
grid.best_score_

Fitting 3 folds for each of 27 candidates, totalling 81 fits
Wall time: 4min 14s


0.0014815729648383718

In [46]:
%%time

print("Best parameters: ", grid_forest.best_params_)
print("Best f1 score: ", grid_forest.best_score_)

Best parameters:  {'max_depth': 1, 'n_estimators': 40}
Best f1 score:  0.0
Wall time: 0 ns


In [47]:
%%time
model_lgb = lgb.LGBMClassifier()
param_grid_lgb = {
    'boosting_type': ['dart'],
    'num_leaves': [100,150],
    'learning_rate': np.arange(0.3, 0.4, 0.1),
    'n_estimators': [100, 150]}
grid_lgb = GridSearchCV(estimator=model_lgb, param_grid=param_grid_lgb, scoring=f1_scorer, cv=2, verbose=2)
grid_lgb.fit(train_tfidf, y_train)

Fitting 2 folds for each of 8 candidates, totalling 16 fits
[CV] END boosting_type=dart, learning_rate=0.3, n_estimators=100, num_leaves=100; total time=  10.0s
[CV] END boosting_type=dart, learning_rate=0.3, n_estimators=100, num_leaves=100; total time=   9.9s
[CV] END boosting_type=dart, learning_rate=0.3, n_estimators=100, num_leaves=150; total time=  13.4s
[CV] END boosting_type=dart, learning_rate=0.3, n_estimators=100, num_leaves=150; total time=  13.4s
[CV] END boosting_type=dart, learning_rate=0.3, n_estimators=150, num_leaves=100; total time=  15.8s
[CV] END boosting_type=dart, learning_rate=0.3, n_estimators=150, num_leaves=100; total time=  15.6s
[CV] END boosting_type=dart, learning_rate=0.3, n_estimators=150, num_leaves=150; total time=  21.9s
[CV] END boosting_type=dart, learning_rate=0.3, n_estimators=150, num_leaves=150; total time=  21.7s
[CV] END boosting_type=dart, learning_rate=0.4, n_estimators=100, num_leaves=100; total time=  10.1s
[CV] END boosting_type=dart, le

GridSearchCV(cv=2, estimator=LGBMClassifier(),
             param_grid={'boosting_type': ['dart'],
                         'learning_rate': array([0.3, 0.4]),
                         'n_estimators': [100, 150], 'num_leaves': [100, 150]},
             scoring=make_scorer(f1_score), verbose=2)

In [48]:
print("Best parameters:", grid_lgb.best_params_)
print("Best f1 score lgb:", grid_lgb.best_score_)

Best parameters: {'boosting_type': 'dart', 'learning_rate': 0.4, 'n_estimators': 100, 'num_leaves': 100}
Best f1 score lgb: 0.7499807300227117


In [62]:
%%time
model_cat = CatBoostClassifier()
param_cat_grid = {'learning_rate': np.arange(0.1, 0.3, 0.1),
                 'depth': [4,6,8],
                 'iterations': [ 100, 500]}

grid_cat = GridSearchCV(estimator=model_cat, param_grid=param_cat_grid, cv=2, verbose=3, scoring=f1_scorer)
grid_cat.fit(train_tfidf, y_train)

Fitting 2 folds for each of 12 candidates, totalling 24 fits
0:	learn: 0.5916222	total: 156ms	remaining: 15.5s
1:	learn: 0.5171360	total: 305ms	remaining: 14.9s
2:	learn: 0.4585792	total: 460ms	remaining: 14.9s
3:	learn: 0.4127875	total: 616ms	remaining: 14.8s
4:	learn: 0.3761455	total: 768ms	remaining: 14.6s
5:	learn: 0.3493415	total: 917ms	remaining: 14.4s
6:	learn: 0.3275737	total: 1.07s	remaining: 14.2s
7:	learn: 0.3098335	total: 1.22s	remaining: 14s
8:	learn: 0.2957930	total: 1.37s	remaining: 13.9s
9:	learn: 0.2827186	total: 1.52s	remaining: 13.7s
10:	learn: 0.2736959	total: 1.68s	remaining: 13.6s
11:	learn: 0.2658746	total: 1.83s	remaining: 13.4s
12:	learn: 0.2591716	total: 1.99s	remaining: 13.3s
13:	learn: 0.2539045	total: 2.14s	remaining: 13.2s
14:	learn: 0.2495839	total: 2.29s	remaining: 13s
15:	learn: 0.2459928	total: 2.45s	remaining: 12.9s
16:	learn: 0.2426379	total: 2.61s	remaining: 12.7s
17:	learn: 0.2396345	total: 2.75s	remaining: 12.6s
18:	learn: 0.2371433	total: 2.91s	r

60:	learn: 0.1980529	total: 9.79s	remaining: 6.26s
61:	learn: 0.1976168	total: 9.95s	remaining: 6.09s
62:	learn: 0.1970925	total: 10.1s	remaining: 5.93s
63:	learn: 0.1965863	total: 10.3s	remaining: 5.77s
64:	learn: 0.1960934	total: 10.4s	remaining: 5.61s
65:	learn: 0.1954763	total: 10.6s	remaining: 5.46s
66:	learn: 0.1948567	total: 10.8s	remaining: 5.3s
67:	learn: 0.1944201	total: 10.9s	remaining: 5.14s
68:	learn: 0.1939349	total: 11.1s	remaining: 4.98s
69:	learn: 0.1933620	total: 11.2s	remaining: 4.82s
70:	learn: 0.1927247	total: 11.4s	remaining: 4.66s
71:	learn: 0.1922604	total: 11.6s	remaining: 4.5s
72:	learn: 0.1919032	total: 11.7s	remaining: 4.33s
73:	learn: 0.1914385	total: 11.9s	remaining: 4.17s
74:	learn: 0.1909064	total: 12s	remaining: 4.01s
75:	learn: 0.1904389	total: 12.2s	remaining: 3.85s
76:	learn: 0.1899424	total: 12.3s	remaining: 3.69s
77:	learn: 0.1892397	total: 12.5s	remaining: 3.53s
78:	learn: 0.1888668	total: 12.7s	remaining: 3.37s
79:	learn: 0.1882788	total: 12.8s	r

20:	learn: 0.2102232	total: 3.28s	remaining: 12.3s
21:	learn: 0.2081122	total: 3.44s	remaining: 12.2s
22:	learn: 0.2066961	total: 3.6s	remaining: 12s
23:	learn: 0.2054752	total: 3.75s	remaining: 11.9s
24:	learn: 0.2034804	total: 3.9s	remaining: 11.7s
25:	learn: 0.2019255	total: 4.05s	remaining: 11.5s
26:	learn: 0.2005848	total: 4.21s	remaining: 11.4s
27:	learn: 0.1992617	total: 4.36s	remaining: 11.2s
28:	learn: 0.1975328	total: 4.52s	remaining: 11.1s
29:	learn: 0.1965658	total: 4.66s	remaining: 10.9s
30:	learn: 0.1954436	total: 4.82s	remaining: 10.7s
31:	learn: 0.1944249	total: 4.96s	remaining: 10.6s
32:	learn: 0.1935559	total: 5.12s	remaining: 10.4s
33:	learn: 0.1923553	total: 5.28s	remaining: 10.2s
34:	learn: 0.1915877	total: 5.42s	remaining: 10.1s
35:	learn: 0.1907585	total: 5.57s	remaining: 9.91s
36:	learn: 0.1898363	total: 5.72s	remaining: 9.75s
37:	learn: 0.1891345	total: 5.88s	remaining: 9.59s
38:	learn: 0.1880695	total: 6.03s	remaining: 9.43s
39:	learn: 0.1872913	total: 6.18s	r

80:	learn: 0.1839571	total: 12.5s	remaining: 1m 4s
81:	learn: 0.1835324	total: 12.6s	remaining: 1m 4s
82:	learn: 0.1831835	total: 12.8s	remaining: 1m 4s
83:	learn: 0.1827040	total: 12.9s	remaining: 1m 4s
84:	learn: 0.1822049	total: 13.1s	remaining: 1m 3s
85:	learn: 0.1818852	total: 13.2s	remaining: 1m 3s
86:	learn: 0.1815524	total: 13.4s	remaining: 1m 3s
87:	learn: 0.1812464	total: 13.6s	remaining: 1m 3s
88:	learn: 0.1809094	total: 13.7s	remaining: 1m 3s
89:	learn: 0.1804753	total: 13.9s	remaining: 1m 3s
90:	learn: 0.1801305	total: 14s	remaining: 1m 3s
91:	learn: 0.1798124	total: 14.2s	remaining: 1m 2s
92:	learn: 0.1795369	total: 14.3s	remaining: 1m 2s
93:	learn: 0.1792549	total: 14.5s	remaining: 1m 2s
94:	learn: 0.1789440	total: 14.6s	remaining: 1m 2s
95:	learn: 0.1786529	total: 14.8s	remaining: 1m 2s
96:	learn: 0.1783066	total: 15s	remaining: 1m 2s
97:	learn: 0.1777632	total: 15.1s	remaining: 1m 2s
98:	learn: 0.1774862	total: 15.3s	remaining: 1m 1s
99:	learn: 0.1771531	total: 15.4s	r

240:	learn: 0.1471976	total: 37s	remaining: 39.7s
241:	learn: 0.1470374	total: 37.2s	remaining: 39.6s
242:	learn: 0.1468527	total: 37.3s	remaining: 39.5s
243:	learn: 0.1467449	total: 37.5s	remaining: 39.3s
244:	learn: 0.1465350	total: 37.6s	remaining: 39.2s
245:	learn: 0.1464634	total: 37.8s	remaining: 39s
246:	learn: 0.1463226	total: 37.9s	remaining: 38.8s
247:	learn: 0.1462085	total: 38.1s	remaining: 38.7s
248:	learn: 0.1459758	total: 38.2s	remaining: 38.5s
249:	learn: 0.1458508	total: 38.4s	remaining: 38.4s
250:	learn: 0.1457256	total: 38.5s	remaining: 38.2s
251:	learn: 0.1456073	total: 38.7s	remaining: 38.1s
252:	learn: 0.1454694	total: 38.9s	remaining: 37.9s
253:	learn: 0.1453694	total: 39s	remaining: 37.8s
254:	learn: 0.1452922	total: 39.2s	remaining: 37.6s
255:	learn: 0.1452171	total: 39.3s	remaining: 37.5s
256:	learn: 0.1450480	total: 39.5s	remaining: 37.3s
257:	learn: 0.1449010	total: 39.6s	remaining: 37.2s
258:	learn: 0.1447606	total: 39.8s	remaining: 37s
259:	learn: 0.144627

400:	learn: 0.1303264	total: 1m 1s	remaining: 15.2s
401:	learn: 0.1301626	total: 1m 1s	remaining: 15s
402:	learn: 0.1300907	total: 1m 1s	remaining: 14.9s
403:	learn: 0.1299924	total: 1m 1s	remaining: 14.7s
404:	learn: 0.1299433	total: 1m 2s	remaining: 14.6s
405:	learn: 0.1298605	total: 1m 2s	remaining: 14.4s
406:	learn: 0.1298124	total: 1m 2s	remaining: 14.2s
407:	learn: 0.1297139	total: 1m 2s	remaining: 14.1s
408:	learn: 0.1296700	total: 1m 2s	remaining: 13.9s
409:	learn: 0.1295496	total: 1m 2s	remaining: 13.8s
410:	learn: 0.1294732	total: 1m 2s	remaining: 13.6s
411:	learn: 0.1293558	total: 1m 3s	remaining: 13.5s
412:	learn: 0.1293000	total: 1m 3s	remaining: 13.3s
413:	learn: 0.1292344	total: 1m 3s	remaining: 13.2s
414:	learn: 0.1290934	total: 1m 3s	remaining: 13s
415:	learn: 0.1289405	total: 1m 3s	remaining: 12.9s
416:	learn: 0.1288085	total: 1m 3s	remaining: 12.7s
417:	learn: 0.1287686	total: 1m 4s	remaining: 12.6s
418:	learn: 0.1287242	total: 1m 4s	remaining: 12.4s
419:	learn: 0.12

56:	learn: 0.2002165	total: 8.96s	remaining: 1m 9s
57:	learn: 0.1997437	total: 9.11s	remaining: 1m 9s
58:	learn: 0.1992995	total: 9.27s	remaining: 1m 9s
59:	learn: 0.1987144	total: 9.41s	remaining: 1m 9s
60:	learn: 0.1980529	total: 9.57s	remaining: 1m 8s
61:	learn: 0.1976168	total: 9.72s	remaining: 1m 8s
62:	learn: 0.1970925	total: 9.88s	remaining: 1m 8s
63:	learn: 0.1965863	total: 10s	remaining: 1m 8s
64:	learn: 0.1960934	total: 10.2s	remaining: 1m 8s
65:	learn: 0.1954763	total: 10.4s	remaining: 1m 8s
66:	learn: 0.1948567	total: 10.5s	remaining: 1m 7s
67:	learn: 0.1944201	total: 10.7s	remaining: 1m 7s
68:	learn: 0.1939349	total: 10.8s	remaining: 1m 7s
69:	learn: 0.1933620	total: 11s	remaining: 1m 7s
70:	learn: 0.1927247	total: 11.1s	remaining: 1m 7s
71:	learn: 0.1922604	total: 11.3s	remaining: 1m 7s
72:	learn: 0.1919032	total: 11.4s	remaining: 1m 6s
73:	learn: 0.1914385	total: 11.6s	remaining: 1m 6s
74:	learn: 0.1909064	total: 11.7s	remaining: 1m 6s
75:	learn: 0.1904389	total: 11.9s	r

216:	learn: 0.1527515	total: 33.6s	remaining: 43.8s
217:	learn: 0.1525308	total: 33.7s	remaining: 43.6s
218:	learn: 0.1523075	total: 33.9s	remaining: 43.5s
219:	learn: 0.1521658	total: 34s	remaining: 43.3s
220:	learn: 0.1518668	total: 34.2s	remaining: 43.2s
221:	learn: 0.1517572	total: 34.3s	remaining: 43s
222:	learn: 0.1516507	total: 34.5s	remaining: 42.9s
223:	learn: 0.1514950	total: 34.6s	remaining: 42.7s
224:	learn: 0.1512489	total: 34.8s	remaining: 42.5s
225:	learn: 0.1510918	total: 34.9s	remaining: 42.4s
226:	learn: 0.1509007	total: 35.1s	remaining: 42.2s
227:	learn: 0.1506133	total: 35.3s	remaining: 42.1s
228:	learn: 0.1504979	total: 35.4s	remaining: 41.9s
229:	learn: 0.1504097	total: 35.6s	remaining: 41.8s
230:	learn: 0.1503092	total: 35.7s	remaining: 41.6s
231:	learn: 0.1501692	total: 35.9s	remaining: 41.4s
232:	learn: 0.1500808	total: 36s	remaining: 41.3s
233:	learn: 0.1499636	total: 36.2s	remaining: 41.1s
234:	learn: 0.1498547	total: 36.3s	remaining: 41s
235:	learn: 0.149562

376:	learn: 0.1340131	total: 58.1s	remaining: 19s
377:	learn: 0.1339637	total: 58.2s	remaining: 18.8s
378:	learn: 0.1338900	total: 58.4s	remaining: 18.6s
379:	learn: 0.1337509	total: 58.6s	remaining: 18.5s
380:	learn: 0.1336782	total: 58.7s	remaining: 18.3s
381:	learn: 0.1336177	total: 58.9s	remaining: 18.2s
382:	learn: 0.1335385	total: 59s	remaining: 18s
383:	learn: 0.1333550	total: 59.2s	remaining: 17.9s
384:	learn: 0.1332696	total: 59.3s	remaining: 17.7s
385:	learn: 0.1332167	total: 59.5s	remaining: 17.6s
386:	learn: 0.1331685	total: 59.6s	remaining: 17.4s
387:	learn: 0.1331227	total: 59.8s	remaining: 17.3s
388:	learn: 0.1330722	total: 59.9s	remaining: 17.1s
389:	learn: 0.1329515	total: 1m	remaining: 16.9s
390:	learn: 0.1327891	total: 1m	remaining: 16.8s
391:	learn: 0.1326686	total: 1m	remaining: 16.6s
392:	learn: 0.1325627	total: 1m	remaining: 16.5s
393:	learn: 0.1325125	total: 1m	remaining: 16.3s
394:	learn: 0.1323591	total: 1m	remaining: 16.2s
395:	learn: 0.1322554	total: 1m 1s	r

34:	learn: 0.1887187	total: 5.4s	remaining: 1m 11s
35:	learn: 0.1878036	total: 5.55s	remaining: 1m 11s
36:	learn: 0.1869293	total: 5.7s	remaining: 1m 11s
37:	learn: 0.1862252	total: 5.84s	remaining: 1m 11s
38:	learn: 0.1855395	total: 5.99s	remaining: 1m 10s
39:	learn: 0.1844327	total: 6.14s	remaining: 1m 10s
40:	learn: 0.1835996	total: 6.3s	remaining: 1m 10s
41:	learn: 0.1828734	total: 6.45s	remaining: 1m 10s
42:	learn: 0.1822133	total: 6.61s	remaining: 1m 10s
43:	learn: 0.1815594	total: 6.78s	remaining: 1m 10s
44:	learn: 0.1807365	total: 6.94s	remaining: 1m 10s
45:	learn: 0.1797317	total: 7.1s	remaining: 1m 10s
46:	learn: 0.1785755	total: 7.25s	remaining: 1m 9s
47:	learn: 0.1779853	total: 7.4s	remaining: 1m 9s
48:	learn: 0.1774668	total: 7.56s	remaining: 1m 9s
49:	learn: 0.1767897	total: 7.71s	remaining: 1m 9s
50:	learn: 0.1759852	total: 7.87s	remaining: 1m 9s
51:	learn: 0.1755345	total: 8.02s	remaining: 1m 9s
52:	learn: 0.1749675	total: 8.18s	remaining: 1m 9s
53:	learn: 0.1743105	tot

194:	learn: 0.1315479	total: 29.8s	remaining: 46.6s
195:	learn: 0.1313540	total: 30s	remaining: 46.5s
196:	learn: 0.1309536	total: 30.1s	remaining: 46.3s
197:	learn: 0.1308577	total: 30.3s	remaining: 46.2s
198:	learn: 0.1307510	total: 30.4s	remaining: 46s
199:	learn: 0.1306116	total: 30.6s	remaining: 45.9s
200:	learn: 0.1303332	total: 30.7s	remaining: 45.7s
201:	learn: 0.1302321	total: 30.9s	remaining: 45.6s
202:	learn: 0.1300483	total: 31s	remaining: 45.4s
203:	learn: 0.1299482	total: 31.2s	remaining: 45.3s
204:	learn: 0.1298741	total: 31.3s	remaining: 45.1s
205:	learn: 0.1296029	total: 31.5s	remaining: 44.9s
206:	learn: 0.1293696	total: 31.7s	remaining: 44.8s
207:	learn: 0.1292274	total: 31.8s	remaining: 44.7s
208:	learn: 0.1291493	total: 32s	remaining: 44.5s
209:	learn: 0.1288137	total: 32.1s	remaining: 44.3s
210:	learn: 0.1286848	total: 32.3s	remaining: 44.2s
211:	learn: 0.1285975	total: 32.4s	remaining: 44s
212:	learn: 0.1284669	total: 32.6s	remaining: 43.9s
213:	learn: 0.1283550	

354:	learn: 0.1106299	total: 54.4s	remaining: 22.2s
355:	learn: 0.1105188	total: 54.5s	remaining: 22.1s
356:	learn: 0.1104539	total: 54.7s	remaining: 21.9s
357:	learn: 0.1102930	total: 54.8s	remaining: 21.7s
358:	learn: 0.1102187	total: 55s	remaining: 21.6s
359:	learn: 0.1101606	total: 55.1s	remaining: 21.4s
360:	learn: 0.1100840	total: 55.3s	remaining: 21.3s
361:	learn: 0.1100303	total: 55.4s	remaining: 21.1s
362:	learn: 0.1099768	total: 55.6s	remaining: 21s
363:	learn: 0.1099119	total: 55.7s	remaining: 20.8s
364:	learn: 0.1098595	total: 55.9s	remaining: 20.7s
365:	learn: 0.1098074	total: 56s	remaining: 20.5s
366:	learn: 0.1096162	total: 56.2s	remaining: 20.4s
367:	learn: 0.1095624	total: 56.3s	remaining: 20.2s
368:	learn: 0.1095070	total: 56.5s	remaining: 20.1s
369:	learn: 0.1093470	total: 56.6s	remaining: 19.9s
370:	learn: 0.1092718	total: 56.8s	remaining: 19.7s
371:	learn: 0.1092123	total: 56.9s	remaining: 19.6s
372:	learn: 0.1091655	total: 57.1s	remaining: 19.4s
373:	learn: 0.1091

12:	learn: 0.2263853	total: 2.01s	remaining: 1m 15s
13:	learn: 0.2235884	total: 2.16s	remaining: 1m 15s
14:	learn: 0.2216331	total: 2.31s	remaining: 1m 14s
15:	learn: 0.2192751	total: 2.47s	remaining: 1m 14s
16:	learn: 0.2171221	total: 2.63s	remaining: 1m 14s
17:	learn: 0.2154702	total: 2.78s	remaining: 1m 14s
18:	learn: 0.2137109	total: 2.93s	remaining: 1m 14s
19:	learn: 0.2122181	total: 3.08s	remaining: 1m 13s
20:	learn: 0.2102232	total: 3.24s	remaining: 1m 13s
21:	learn: 0.2081122	total: 3.39s	remaining: 1m 13s
22:	learn: 0.2066961	total: 3.55s	remaining: 1m 13s
23:	learn: 0.2054752	total: 3.7s	remaining: 1m 13s
24:	learn: 0.2034804	total: 3.85s	remaining: 1m 13s
25:	learn: 0.2019255	total: 4.01s	remaining: 1m 13s
26:	learn: 0.2005848	total: 4.17s	remaining: 1m 12s
27:	learn: 0.1992617	total: 4.33s	remaining: 1m 13s
28:	learn: 0.1975328	total: 4.5s	remaining: 1m 13s
29:	learn: 0.1965658	total: 4.65s	remaining: 1m 12s
30:	learn: 0.1954436	total: 4.8s	remaining: 1m 12s
31:	learn: 0.19

172:	learn: 0.1363894	total: 26.7s	remaining: 50.4s
173:	learn: 0.1361702	total: 26.8s	remaining: 50.3s
174:	learn: 0.1360499	total: 27s	remaining: 50.1s
175:	learn: 0.1359430	total: 27.1s	remaining: 49.9s
176:	learn: 0.1358268	total: 27.3s	remaining: 49.8s
177:	learn: 0.1356147	total: 27.4s	remaining: 49.6s
178:	learn: 0.1353484	total: 27.6s	remaining: 49.5s
179:	learn: 0.1350586	total: 27.8s	remaining: 49.3s
180:	learn: 0.1349454	total: 27.9s	remaining: 49.2s
181:	learn: 0.1348197	total: 28.1s	remaining: 49s
182:	learn: 0.1344859	total: 28.2s	remaining: 48.9s
183:	learn: 0.1342811	total: 28.4s	remaining: 48.7s
184:	learn: 0.1341649	total: 28.5s	remaining: 48.6s
185:	learn: 0.1340735	total: 28.7s	remaining: 48.4s
186:	learn: 0.1338098	total: 28.8s	remaining: 48.2s
187:	learn: 0.1336998	total: 29s	remaining: 48.1s
188:	learn: 0.1333051	total: 29.1s	remaining: 47.9s
189:	learn: 0.1332121	total: 29.3s	remaining: 47.8s
190:	learn: 0.1329068	total: 29.4s	remaining: 47.6s
191:	learn: 0.1328

332:	learn: 0.1145762	total: 51s	remaining: 25.6s
333:	learn: 0.1145160	total: 51.2s	remaining: 25.4s
334:	learn: 0.1144195	total: 51.3s	remaining: 25.3s
335:	learn: 0.1142739	total: 51.5s	remaining: 25.1s
336:	learn: 0.1142072	total: 51.6s	remaining: 25s
337:	learn: 0.1140450	total: 51.8s	remaining: 24.8s
338:	learn: 0.1139691	total: 51.9s	remaining: 24.7s
339:	learn: 0.1138086	total: 52.1s	remaining: 24.5s
340:	learn: 0.1137600	total: 52.2s	remaining: 24.4s
341:	learn: 0.1136116	total: 52.4s	remaining: 24.2s
342:	learn: 0.1135119	total: 52.6s	remaining: 24.1s
343:	learn: 0.1134461	total: 52.7s	remaining: 23.9s
344:	learn: 0.1133204	total: 52.9s	remaining: 23.8s
345:	learn: 0.1132270	total: 53s	remaining: 23.6s
346:	learn: 0.1131047	total: 53.2s	remaining: 23.4s
347:	learn: 0.1130410	total: 53.3s	remaining: 23.3s
348:	learn: 0.1129780	total: 53.5s	remaining: 23.1s
349:	learn: 0.1128541	total: 53.6s	remaining: 23s
350:	learn: 0.1127977	total: 53.8s	remaining: 22.8s
351:	learn: 0.112670

490:	learn: 0.1012828	total: 1m 15s	remaining: 1.38s
491:	learn: 0.1012326	total: 1m 15s	remaining: 1.22s
492:	learn: 0.1011846	total: 1m 15s	remaining: 1.07s
493:	learn: 0.1010883	total: 1m 15s	remaining: 918ms
494:	learn: 0.1010407	total: 1m 15s	remaining: 765ms
495:	learn: 0.1009195	total: 1m 15s	remaining: 612ms
496:	learn: 0.1008597	total: 1m 16s	remaining: 459ms
497:	learn: 0.1008056	total: 1m 16s	remaining: 306ms
498:	learn: 0.1007593	total: 1m 16s	remaining: 153ms
499:	learn: 0.1006419	total: 1m 16s	remaining: 0us
[CV 2/2] END depth=4, iterations=500, learning_rate=0.2;, score=0.719 total time= 1.3min
0:	learn: 0.5915684	total: 326ms	remaining: 32.3s
1:	learn: 0.5134455	total: 655ms	remaining: 32.1s
2:	learn: 0.4549021	total: 975ms	remaining: 31.5s
3:	learn: 0.4091333	total: 1.3s	remaining: 31.3s
4:	learn: 0.3711497	total: 1.64s	remaining: 31.2s
5:	learn: 0.3430473	total: 1.98s	remaining: 30.9s
6:	learn: 0.3204016	total: 2.29s	remaining: 30.5s
7:	learn: 0.3023377	total: 2.62s	r

49:	learn: 0.1915331	total: 16.2s	remaining: 16.2s
50:	learn: 0.1906765	total: 16.5s	remaining: 15.9s
51:	learn: 0.1897590	total: 16.9s	remaining: 15.6s
52:	learn: 0.1890222	total: 17.2s	remaining: 15.3s
53:	learn: 0.1883826	total: 17.5s	remaining: 14.9s
54:	learn: 0.1875847	total: 17.8s	remaining: 14.6s
55:	learn: 0.1867748	total: 18.2s	remaining: 14.3s
56:	learn: 0.1862107	total: 18.5s	remaining: 13.9s
57:	learn: 0.1857214	total: 18.8s	remaining: 13.6s
58:	learn: 0.1851751	total: 19.2s	remaining: 13.3s
59:	learn: 0.1846907	total: 19.5s	remaining: 13s
60:	learn: 0.1840441	total: 19.8s	remaining: 12.7s
61:	learn: 0.1835663	total: 20.2s	remaining: 12.4s
62:	learn: 0.1830751	total: 20.5s	remaining: 12s
63:	learn: 0.1826180	total: 20.8s	remaining: 11.7s
64:	learn: 0.1818153	total: 21.2s	remaining: 11.4s
65:	learn: 0.1811195	total: 21.5s	remaining: 11.1s
66:	learn: 0.1806113	total: 21.9s	remaining: 10.8s
67:	learn: 0.1801394	total: 22.2s	remaining: 10.4s
68:	learn: 0.1797115	total: 22.5s	r

8:	learn: 0.2301432	total: 2.94s	remaining: 29.8s
9:	learn: 0.2253038	total: 3.26s	remaining: 29.3s
10:	learn: 0.2214276	total: 3.58s	remaining: 29s
11:	learn: 0.2166802	total: 3.91s	remaining: 28.7s
12:	learn: 0.2137684	total: 4.23s	remaining: 28.3s
13:	learn: 0.2110785	total: 4.55s	remaining: 27.9s
14:	learn: 0.2085673	total: 4.86s	remaining: 27.6s
15:	learn: 0.2063280	total: 5.19s	remaining: 27.2s
16:	learn: 0.2047104	total: 5.51s	remaining: 26.9s
17:	learn: 0.2029425	total: 5.84s	remaining: 26.6s
18:	learn: 0.2006435	total: 6.19s	remaining: 26.4s
19:	learn: 0.1985591	total: 6.51s	remaining: 26s
20:	learn: 0.1967900	total: 6.83s	remaining: 25.7s
21:	learn: 0.1946827	total: 7.15s	remaining: 25.3s
22:	learn: 0.1930049	total: 7.47s	remaining: 25s
23:	learn: 0.1916021	total: 7.81s	remaining: 24.7s
24:	learn: 0.1904080	total: 8.13s	remaining: 24.4s
25:	learn: 0.1888560	total: 8.46s	remaining: 24.1s
26:	learn: 0.1879157	total: 8.77s	remaining: 23.7s
27:	learn: 0.1867455	total: 9.09s	remai

67:	learn: 0.1775731	total: 22s	remaining: 2m 19s
68:	learn: 0.1770999	total: 22.3s	remaining: 2m 19s
69:	learn: 0.1766884	total: 22.6s	remaining: 2m 19s
70:	learn: 0.1761736	total: 23s	remaining: 2m 18s
71:	learn: 0.1758154	total: 23.3s	remaining: 2m 18s
72:	learn: 0.1752868	total: 23.6s	remaining: 2m 17s
73:	learn: 0.1748234	total: 23.9s	remaining: 2m 17s
74:	learn: 0.1744411	total: 24.2s	remaining: 2m 17s
75:	learn: 0.1739089	total: 24.5s	remaining: 2m 16s
76:	learn: 0.1735318	total: 24.9s	remaining: 2m 16s
77:	learn: 0.1731878	total: 25.2s	remaining: 2m 16s
78:	learn: 0.1727747	total: 25.5s	remaining: 2m 15s
79:	learn: 0.1722785	total: 25.8s	remaining: 2m 15s
80:	learn: 0.1719635	total: 26.2s	remaining: 2m 15s
81:	learn: 0.1715306	total: 26.5s	remaining: 2m 15s
82:	learn: 0.1710935	total: 26.8s	remaining: 2m 14s
83:	learn: 0.1707422	total: 27.1s	remaining: 2m 14s
84:	learn: 0.1704047	total: 27.4s	remaining: 2m 13s
85:	learn: 0.1699768	total: 27.8s	remaining: 2m 13s
86:	learn: 0.169

224:	learn: 0.1355507	total: 1m 12s	remaining: 1m 28s
225:	learn: 0.1353697	total: 1m 12s	remaining: 1m 28s
226:	learn: 0.1351898	total: 1m 13s	remaining: 1m 27s
227:	learn: 0.1350108	total: 1m 13s	remaining: 1m 27s
228:	learn: 0.1347881	total: 1m 13s	remaining: 1m 27s
229:	learn: 0.1346214	total: 1m 14s	remaining: 1m 26s
230:	learn: 0.1345398	total: 1m 14s	remaining: 1m 26s
231:	learn: 0.1344577	total: 1m 14s	remaining: 1m 26s
232:	learn: 0.1342902	total: 1m 15s	remaining: 1m 25s
233:	learn: 0.1342119	total: 1m 15s	remaining: 1m 25s
234:	learn: 0.1340124	total: 1m 15s	remaining: 1m 25s
235:	learn: 0.1339167	total: 1m 15s	remaining: 1m 24s
236:	learn: 0.1337168	total: 1m 16s	remaining: 1m 24s
237:	learn: 0.1335817	total: 1m 16s	remaining: 1m 24s
238:	learn: 0.1334602	total: 1m 16s	remaining: 1m 24s
239:	learn: 0.1333808	total: 1m 17s	remaining: 1m 23s
240:	learn: 0.1332047	total: 1m 17s	remaining: 1m 23s
241:	learn: 0.1331305	total: 1m 17s	remaining: 1m 23s
242:	learn: 0.1330592	total:

379:	learn: 0.1167664	total: 2m 1s	remaining: 38.4s
380:	learn: 0.1167152	total: 2m 2s	remaining: 38.1s
381:	learn: 0.1166479	total: 2m 2s	remaining: 37.8s
382:	learn: 0.1166000	total: 2m 2s	remaining: 37.5s
383:	learn: 0.1164973	total: 2m 2s	remaining: 37.1s
384:	learn: 0.1164479	total: 2m 3s	remaining: 36.8s
385:	learn: 0.1163410	total: 2m 3s	remaining: 36.5s
386:	learn: 0.1162278	total: 2m 3s	remaining: 36.2s
387:	learn: 0.1161594	total: 2m 4s	remaining: 35.9s
388:	learn: 0.1161086	total: 2m 4s	remaining: 35.5s
389:	learn: 0.1159368	total: 2m 4s	remaining: 35.2s
390:	learn: 0.1158068	total: 2m 5s	remaining: 34.9s
391:	learn: 0.1156644	total: 2m 5s	remaining: 34.6s
392:	learn: 0.1155695	total: 2m 5s	remaining: 34.3s
393:	learn: 0.1154547	total: 2m 6s	remaining: 33.9s
394:	learn: 0.1153697	total: 2m 6s	remaining: 33.6s
395:	learn: 0.1152581	total: 2m 6s	remaining: 33.3s
396:	learn: 0.1151610	total: 2m 7s	remaining: 33s
397:	learn: 0.1150692	total: 2m 7s	remaining: 32.6s
398:	learn: 0.

34:	learn: 0.2034746	total: 11.1s	remaining: 2m 27s
35:	learn: 0.2024911	total: 11.4s	remaining: 2m 27s
36:	learn: 0.2014691	total: 11.7s	remaining: 2m 26s
37:	learn: 0.2004820	total: 12.1s	remaining: 2m 26s
38:	learn: 0.1997431	total: 12.4s	remaining: 2m 26s
39:	learn: 0.1989543	total: 12.7s	remaining: 2m 25s
40:	learn: 0.1980586	total: 13s	remaining: 2m 25s
41:	learn: 0.1973444	total: 13.3s	remaining: 2m 24s
42:	learn: 0.1966701	total: 13.6s	remaining: 2m 24s
43:	learn: 0.1960263	total: 13.9s	remaining: 2m 24s
44:	learn: 0.1953958	total: 14.2s	remaining: 2m 23s
45:	learn: 0.1946137	total: 14.5s	remaining: 2m 23s
46:	learn: 0.1935745	total: 14.8s	remaining: 2m 22s
47:	learn: 0.1929146	total: 15.1s	remaining: 2m 22s
48:	learn: 0.1922814	total: 15.4s	remaining: 2m 22s
49:	learn: 0.1915331	total: 15.7s	remaining: 2m 21s
50:	learn: 0.1906765	total: 16.1s	remaining: 2m 21s
51:	learn: 0.1897590	total: 16.4s	remaining: 2m 21s
52:	learn: 0.1890222	total: 16.7s	remaining: 2m 20s
53:	learn: 0.1

192:	learn: 0.1419088	total: 1m	remaining: 1m 36s
193:	learn: 0.1417807	total: 1m	remaining: 1m 35s
194:	learn: 0.1416569	total: 1m 1s	remaining: 1m 35s
195:	learn: 0.1413403	total: 1m 1s	remaining: 1m 35s
196:	learn: 0.1411878	total: 1m 1s	remaining: 1m 35s
197:	learn: 0.1410422	total: 1m 2s	remaining: 1m 34s
198:	learn: 0.1408690	total: 1m 2s	remaining: 1m 34s
199:	learn: 0.1407474	total: 1m 2s	remaining: 1m 34s
200:	learn: 0.1405509	total: 1m 2s	remaining: 1m 33s
201:	learn: 0.1403624	total: 1m 3s	remaining: 1m 33s
202:	learn: 0.1401544	total: 1m 3s	remaining: 1m 33s
203:	learn: 0.1399869	total: 1m 3s	remaining: 1m 32s
204:	learn: 0.1398056	total: 1m 4s	remaining: 1m 32s
205:	learn: 0.1395913	total: 1m 4s	remaining: 1m 32s
206:	learn: 0.1393971	total: 1m 4s	remaining: 1m 31s
207:	learn: 0.1392209	total: 1m 5s	remaining: 1m 31s
208:	learn: 0.1391150	total: 1m 5s	remaining: 1m 31s
209:	learn: 0.1389673	total: 1m 5s	remaining: 1m 30s
210:	learn: 0.1388595	total: 1m 6s	remaining: 1m 30s

347:	learn: 0.1218146	total: 1m 48s	remaining: 47.5s
348:	learn: 0.1217518	total: 1m 49s	remaining: 47.2s
349:	learn: 0.1216633	total: 1m 49s	remaining: 46.8s
350:	learn: 0.1214916	total: 1m 49s	remaining: 46.5s
351:	learn: 0.1213646	total: 1m 49s	remaining: 46.2s
352:	learn: 0.1211894	total: 1m 50s	remaining: 45.9s
353:	learn: 0.1210705	total: 1m 50s	remaining: 45.6s
354:	learn: 0.1208477	total: 1m 50s	remaining: 45.3s
355:	learn: 0.1207927	total: 1m 51s	remaining: 45s
356:	learn: 0.1206078	total: 1m 51s	remaining: 44.6s
357:	learn: 0.1205088	total: 1m 51s	remaining: 44.3s
358:	learn: 0.1204432	total: 1m 52s	remaining: 44s
359:	learn: 0.1203785	total: 1m 52s	remaining: 43.7s
360:	learn: 0.1203239	total: 1m 52s	remaining: 43.4s
361:	learn: 0.1201995	total: 1m 53s	remaining: 43.1s
362:	learn: 0.1200552	total: 1m 53s	remaining: 42.8s
363:	learn: 0.1199981	total: 1m 53s	remaining: 42.5s
364:	learn: 0.1199236	total: 1m 53s	remaining: 42.2s
365:	learn: 0.1197771	total: 1m 54s	remaining: 41.

2:	learn: 0.3356465	total: 937ms	remaining: 2m 35s
3:	learn: 0.2938927	total: 1.25s	remaining: 2m 34s
4:	learn: 0.2689352	total: 1.57s	remaining: 2m 35s
5:	learn: 0.2517193	total: 1.88s	remaining: 2m 34s
6:	learn: 0.2404009	total: 2.19s	remaining: 2m 34s
7:	learn: 0.2323483	total: 2.5s	remaining: 2m 33s
8:	learn: 0.2258283	total: 2.81s	remaining: 2m 33s
9:	learn: 0.2211731	total: 3.15s	remaining: 2m 34s
10:	learn: 0.2167625	total: 3.47s	remaining: 2m 34s
11:	learn: 0.2137179	total: 3.78s	remaining: 2m 33s
12:	learn: 0.2098094	total: 4.09s	remaining: 2m 33s
13:	learn: 0.2074688	total: 4.4s	remaining: 2m 32s
14:	learn: 0.2045033	total: 4.74s	remaining: 2m 33s
15:	learn: 0.2020089	total: 5.05s	remaining: 2m 32s
16:	learn: 0.1992387	total: 5.36s	remaining: 2m 32s
17:	learn: 0.1969744	total: 5.67s	remaining: 2m 31s
18:	learn: 0.1951971	total: 5.97s	remaining: 2m 31s
19:	learn: 0.1938674	total: 6.28s	remaining: 2m 30s
20:	learn: 0.1925150	total: 6.59s	remaining: 2m 30s
21:	learn: 0.1913655	t

160:	learn: 0.1221006	total: 50.5s	remaining: 1m 46s
161:	learn: 0.1218272	total: 50.8s	remaining: 1m 45s
162:	learn: 0.1217125	total: 51.1s	remaining: 1m 45s
163:	learn: 0.1213899	total: 51.4s	remaining: 1m 45s
164:	learn: 0.1210771	total: 51.7s	remaining: 1m 45s
165:	learn: 0.1208140	total: 52s	remaining: 1m 44s
166:	learn: 0.1205660	total: 52.4s	remaining: 1m 44s
167:	learn: 0.1202107	total: 52.7s	remaining: 1m 44s
168:	learn: 0.1200907	total: 53s	remaining: 1m 43s
169:	learn: 0.1197919	total: 53.3s	remaining: 1m 43s
170:	learn: 0.1194574	total: 53.6s	remaining: 1m 43s
171:	learn: 0.1193295	total: 53.9s	remaining: 1m 42s
172:	learn: 0.1189806	total: 54.2s	remaining: 1m 42s
173:	learn: 0.1188537	total: 54.5s	remaining: 1m 42s
174:	learn: 0.1186351	total: 54.8s	remaining: 1m 41s
175:	learn: 0.1184777	total: 55.1s	remaining: 1m 41s
176:	learn: 0.1181912	total: 55.5s	remaining: 1m 41s
177:	learn: 0.1179412	total: 55.8s	remaining: 1m 40s
178:	learn: 0.1177793	total: 56.1s	remaining: 1m 4

315:	learn: 0.0992001	total: 1m 38s	remaining: 57.4s
316:	learn: 0.0991366	total: 1m 38s	remaining: 57.1s
317:	learn: 0.0990696	total: 1m 39s	remaining: 56.7s
318:	learn: 0.0990023	total: 1m 39s	remaining: 56.4s
319:	learn: 0.0988083	total: 1m 39s	remaining: 56.1s
320:	learn: 0.0987432	total: 1m 40s	remaining: 55.8s
321:	learn: 0.0986767	total: 1m 40s	remaining: 55.5s
322:	learn: 0.0986102	total: 1m 40s	remaining: 55.2s
323:	learn: 0.0985440	total: 1m 41s	remaining: 54.9s
324:	learn: 0.0984853	total: 1m 41s	remaining: 54.6s
325:	learn: 0.0983041	total: 1m 41s	remaining: 54.3s
326:	learn: 0.0982086	total: 1m 41s	remaining: 53.9s
327:	learn: 0.0981427	total: 1m 42s	remaining: 53.6s
328:	learn: 0.0980783	total: 1m 42s	remaining: 53.3s
329:	learn: 0.0980141	total: 1m 42s	remaining: 53s
330:	learn: 0.0978579	total: 1m 43s	remaining: 52.7s
331:	learn: 0.0977922	total: 1m 43s	remaining: 52.4s
332:	learn: 0.0977182	total: 1m 43s	remaining: 52.1s
333:	learn: 0.0976525	total: 1m 44s	remaining: 5

471:	learn: 0.0866317	total: 2m 26s	remaining: 8.71s
472:	learn: 0.0864522	total: 2m 27s	remaining: 8.4s
473:	learn: 0.0864045	total: 2m 27s	remaining: 8.09s
474:	learn: 0.0863555	total: 2m 27s	remaining: 7.78s
475:	learn: 0.0863057	total: 2m 28s	remaining: 7.46s
476:	learn: 0.0862439	total: 2m 28s	remaining: 7.16s
477:	learn: 0.0861915	total: 2m 28s	remaining: 6.84s
478:	learn: 0.0861205	total: 2m 29s	remaining: 6.53s
479:	learn: 0.0860724	total: 2m 29s	remaining: 6.22s
480:	learn: 0.0859284	total: 2m 29s	remaining: 5.91s
481:	learn: 0.0858828	total: 2m 29s	remaining: 5.6s
482:	learn: 0.0858364	total: 2m 30s	remaining: 5.29s
483:	learn: 0.0856421	total: 2m 30s	remaining: 4.98s
484:	learn: 0.0855894	total: 2m 30s	remaining: 4.67s
485:	learn: 0.0855403	total: 2m 31s	remaining: 4.36s
486:	learn: 0.0854919	total: 2m 31s	remaining: 4.04s
487:	learn: 0.0853302	total: 2m 31s	remaining: 3.73s
488:	learn: 0.0852732	total: 2m 32s	remaining: 3.42s
489:	learn: 0.0852258	total: 2m 32s	remaining: 3

128:	learn: 0.1325253	total: 40.1s	remaining: 1m 55s
129:	learn: 0.1321683	total: 40.4s	remaining: 1m 55s
130:	learn: 0.1320140	total: 40.8s	remaining: 1m 54s
131:	learn: 0.1317360	total: 41.1s	remaining: 1m 54s
132:	learn: 0.1315284	total: 41.4s	remaining: 1m 54s
133:	learn: 0.1312740	total: 41.7s	remaining: 1m 53s
134:	learn: 0.1311023	total: 42s	remaining: 1m 53s
135:	learn: 0.1307833	total: 42.3s	remaining: 1m 53s
136:	learn: 0.1306389	total: 42.6s	remaining: 1m 52s
137:	learn: 0.1304828	total: 42.9s	remaining: 1m 52s
138:	learn: 0.1303103	total: 43.3s	remaining: 1m 52s
139:	learn: 0.1299090	total: 43.6s	remaining: 1m 52s
140:	learn: 0.1297611	total: 43.9s	remaining: 1m 51s
141:	learn: 0.1292584	total: 44.2s	remaining: 1m 51s
142:	learn: 0.1290972	total: 44.5s	remaining: 1m 51s
143:	learn: 0.1287611	total: 44.9s	remaining: 1m 50s
144:	learn: 0.1283396	total: 45.2s	remaining: 1m 50s
145:	learn: 0.1280637	total: 45.5s	remaining: 1m 50s
146:	learn: 0.1275529	total: 45.8s	remaining: 1m

282:	learn: 0.1031614	total: 1m 30s	remaining: 1m 9s
283:	learn: 0.1030221	total: 1m 30s	remaining: 1m 9s
284:	learn: 0.1029469	total: 1m 31s	remaining: 1m 8s
285:	learn: 0.1028761	total: 1m 31s	remaining: 1m 8s
286:	learn: 0.1028008	total: 1m 31s	remaining: 1m 8s
287:	learn: 0.1027284	total: 1m 32s	remaining: 1m 7s
288:	learn: 0.1024365	total: 1m 32s	remaining: 1m 7s
289:	learn: 0.1023676	total: 1m 32s	remaining: 1m 7s
290:	learn: 0.1022875	total: 1m 33s	remaining: 1m 6s
291:	learn: 0.1022124	total: 1m 33s	remaining: 1m 6s
292:	learn: 0.1021206	total: 1m 33s	remaining: 1m 6s
293:	learn: 0.1020542	total: 1m 34s	remaining: 1m 5s
294:	learn: 0.1019609	total: 1m 34s	remaining: 1m 5s
295:	learn: 0.1018885	total: 1m 34s	remaining: 1m 5s
296:	learn: 0.1018197	total: 1m 35s	remaining: 1m 4s
297:	learn: 0.1017266	total: 1m 35s	remaining: 1m 4s
298:	learn: 0.1015857	total: 1m 35s	remaining: 1m 4s
299:	learn: 0.1015133	total: 1m 36s	remaining: 1m 4s
300:	learn: 0.1014436	total: 1m 36s	remaining:

439:	learn: 0.0893102	total: 2m 22s	remaining: 19.4s
440:	learn: 0.0892577	total: 2m 22s	remaining: 19.1s
441:	learn: 0.0892036	total: 2m 22s	remaining: 18.7s
442:	learn: 0.0891519	total: 2m 23s	remaining: 18.4s
443:	learn: 0.0890985	total: 2m 23s	remaining: 18.1s
444:	learn: 0.0889585	total: 2m 23s	remaining: 17.8s
445:	learn: 0.0889052	total: 2m 24s	remaining: 17.4s
446:	learn: 0.0888533	total: 2m 24s	remaining: 17.1s
447:	learn: 0.0887895	total: 2m 24s	remaining: 16.8s
448:	learn: 0.0887368	total: 2m 24s	remaining: 16.5s
449:	learn: 0.0886841	total: 2m 25s	remaining: 16.1s
450:	learn: 0.0886318	total: 2m 25s	remaining: 15.8s
451:	learn: 0.0884800	total: 2m 25s	remaining: 15.5s
452:	learn: 0.0884290	total: 2m 26s	remaining: 15.2s
453:	learn: 0.0883769	total: 2m 26s	remaining: 14.8s
454:	learn: 0.0883260	total: 2m 26s	remaining: 14.5s
455:	learn: 0.0882741	total: 2m 27s	remaining: 14.2s
456:	learn: 0.0882226	total: 2m 27s	remaining: 13.9s
457:	learn: 0.0880674	total: 2m 27s	remaining:

96:	learn: 0.1560565	total: 1m 37s	remaining: 3s
97:	learn: 0.1557028	total: 1m 38s	remaining: 2s
98:	learn: 0.1553526	total: 1m 39s	remaining: 1s
99:	learn: 0.1549647	total: 1m 39s	remaining: 0us
[CV 1/2] END depth=8, iterations=100, learning_rate=0.1;, score=0.646 total time= 1.7min
0:	learn: 0.5976575	total: 979ms	remaining: 1m 36s
1:	learn: 0.5235431	total: 1.92s	remaining: 1m 34s
2:	learn: 0.4644302	total: 2.88s	remaining: 1m 33s
3:	learn: 0.4145482	total: 3.91s	remaining: 1m 33s
4:	learn: 0.3756498	total: 4.96s	remaining: 1m 34s
5:	learn: 0.3455259	total: 6s	remaining: 1m 34s
6:	learn: 0.3225351	total: 7.07s	remaining: 1m 33s
7:	learn: 0.3040743	total: 8.04s	remaining: 1m 32s
8:	learn: 0.2884702	total: 8.98s	remaining: 1m 30s
9:	learn: 0.2765902	total: 9.93s	remaining: 1m 29s
10:	learn: 0.2665324	total: 10.9s	remaining: 1m 28s
11:	learn: 0.2568379	total: 12s	remaining: 1m 27s
12:	learn: 0.2497413	total: 13s	remaining: 1m 27s
13:	learn: 0.2441922	total: 14s	remaining: 1m 25s
14:	l

54:	learn: 0.1514506	total: 52.6s	remaining: 43s
55:	learn: 0.1509375	total: 53.5s	remaining: 42s
56:	learn: 0.1501258	total: 54.4s	remaining: 41.1s
57:	learn: 0.1497130	total: 55.4s	remaining: 40.1s
58:	learn: 0.1489441	total: 56.4s	remaining: 39.2s
59:	learn: 0.1483422	total: 57.3s	remaining: 38.2s
60:	learn: 0.1477609	total: 58.3s	remaining: 37.2s
61:	learn: 0.1471200	total: 59.2s	remaining: 36.3s
62:	learn: 0.1467056	total: 1m	remaining: 35.3s
63:	learn: 0.1461441	total: 1m 1s	remaining: 34.3s
64:	learn: 0.1451324	total: 1m 2s	remaining: 33.4s
65:	learn: 0.1443646	total: 1m 2s	remaining: 32.4s
66:	learn: 0.1438780	total: 1m 3s	remaining: 31.5s
67:	learn: 0.1431817	total: 1m 4s	remaining: 30.5s
68:	learn: 0.1426392	total: 1m 5s	remaining: 29.6s
69:	learn: 0.1420683	total: 1m 6s	remaining: 28.6s
70:	learn: 0.1417026	total: 1m 7s	remaining: 27.6s
71:	learn: 0.1411209	total: 1m 8s	remaining: 26.7s
72:	learn: 0.1407981	total: 1m 9s	remaining: 25.7s
73:	learn: 0.1401196	total: 1m 10s	rem

11:	learn: 0.2507200	total: 11.5s	remaining: 7m 46s
12:	learn: 0.2444868	total: 12.4s	remaining: 7m 44s
13:	learn: 0.2384906	total: 13.3s	remaining: 7m 43s
14:	learn: 0.2335628	total: 14.3s	remaining: 7m 41s
15:	learn: 0.2291596	total: 15.3s	remaining: 7m 43s
16:	learn: 0.2256337	total: 16.3s	remaining: 7m 44s
17:	learn: 0.2222126	total: 17.4s	remaining: 7m 46s
18:	learn: 0.2192844	total: 18.5s	remaining: 7m 48s
19:	learn: 0.2159267	total: 19.5s	remaining: 7m 48s
20:	learn: 0.2135908	total: 20.5s	remaining: 7m 46s
21:	learn: 0.2114497	total: 21.4s	remaining: 7m 45s
22:	learn: 0.2096477	total: 22.4s	remaining: 7m 45s
23:	learn: 0.2077183	total: 23.4s	remaining: 7m 43s
24:	learn: 0.2054403	total: 24.3s	remaining: 7m 42s
25:	learn: 0.2038770	total: 25.3s	remaining: 7m 40s
26:	learn: 0.2022767	total: 26.2s	remaining: 7m 38s
27:	learn: 0.2008487	total: 27.1s	remaining: 7m 37s
28:	learn: 0.1992197	total: 28.1s	remaining: 7m 36s
29:	learn: 0.1980023	total: 29.1s	remaining: 7m 35s
30:	learn: 0

167:	learn: 0.1352406	total: 2m 42s	remaining: 5m 21s
168:	learn: 0.1351029	total: 2m 43s	remaining: 5m 20s
169:	learn: 0.1349218	total: 2m 44s	remaining: 5m 19s
170:	learn: 0.1347904	total: 2m 45s	remaining: 5m 18s
171:	learn: 0.1345558	total: 2m 46s	remaining: 5m 17s
172:	learn: 0.1343096	total: 2m 47s	remaining: 5m 16s
173:	learn: 0.1340972	total: 2m 48s	remaining: 5m 15s
174:	learn: 0.1337678	total: 2m 49s	remaining: 5m 14s
175:	learn: 0.1335079	total: 2m 50s	remaining: 5m 13s
176:	learn: 0.1333298	total: 2m 51s	remaining: 5m 12s
177:	learn: 0.1331798	total: 2m 52s	remaining: 5m 11s
178:	learn: 0.1330631	total: 2m 53s	remaining: 5m 10s
179:	learn: 0.1328319	total: 2m 54s	remaining: 5m 9s
180:	learn: 0.1325383	total: 2m 55s	remaining: 5m 8s
181:	learn: 0.1324278	total: 2m 55s	remaining: 5m 7s
182:	learn: 0.1321263	total: 2m 56s	remaining: 5m 6s
183:	learn: 0.1319317	total: 2m 57s	remaining: 5m 5s
184:	learn: 0.1315707	total: 2m 58s	remaining: 5m 4s
185:	learn: 0.1314640	total: 2m 59

321:	learn: 0.1101430	total: 5m 10s	remaining: 2m 51s
322:	learn: 0.1100489	total: 5m 11s	remaining: 2m 50s
323:	learn: 0.1099179	total: 5m 12s	remaining: 2m 49s
324:	learn: 0.1098538	total: 5m 13s	remaining: 2m 48s
325:	learn: 0.1097023	total: 5m 14s	remaining: 2m 47s
326:	learn: 0.1096479	total: 5m 15s	remaining: 2m 46s
327:	learn: 0.1095318	total: 5m 16s	remaining: 2m 45s
328:	learn: 0.1093505	total: 5m 17s	remaining: 2m 44s
329:	learn: 0.1091927	total: 5m 18s	remaining: 2m 44s
330:	learn: 0.1090918	total: 5m 19s	remaining: 2m 43s
331:	learn: 0.1089531	total: 5m 20s	remaining: 2m 42s
332:	learn: 0.1087546	total: 5m 21s	remaining: 2m 41s
333:	learn: 0.1086956	total: 5m 22s	remaining: 2m 40s
334:	learn: 0.1085629	total: 5m 23s	remaining: 2m 39s
335:	learn: 0.1084002	total: 5m 24s	remaining: 2m 38s
336:	learn: 0.1082099	total: 5m 25s	remaining: 2m 37s
337:	learn: 0.1081558	total: 5m 26s	remaining: 2m 36s
338:	learn: 0.1081027	total: 5m 27s	remaining: 2m 35s
339:	learn: 0.1080503	total:

475:	learn: 0.0986260	total: 7m 39s	remaining: 23.2s
476:	learn: 0.0985841	total: 7m 40s	remaining: 22.2s
477:	learn: 0.0985433	total: 7m 41s	remaining: 21.2s
478:	learn: 0.0985018	total: 7m 42s	remaining: 20.3s
479:	learn: 0.0984607	total: 7m 43s	remaining: 19.3s
480:	learn: 0.0981496	total: 7m 43s	remaining: 18.3s
481:	learn: 0.0981081	total: 7m 44s	remaining: 17.4s
482:	learn: 0.0980678	total: 7m 45s	remaining: 16.4s
483:	learn: 0.0979931	total: 7m 46s	remaining: 15.4s
484:	learn: 0.0979533	total: 7m 47s	remaining: 14.5s
485:	learn: 0.0979135	total: 7m 48s	remaining: 13.5s
486:	learn: 0.0977974	total: 7m 49s	remaining: 12.5s
487:	learn: 0.0977565	total: 7m 50s	remaining: 11.6s
488:	learn: 0.0977154	total: 7m 51s	remaining: 10.6s
489:	learn: 0.0975101	total: 7m 52s	remaining: 9.64s
490:	learn: 0.0974326	total: 7m 53s	remaining: 8.68s
491:	learn: 0.0973914	total: 7m 54s	remaining: 7.71s
492:	learn: 0.0973503	total: 7m 55s	remaining: 6.75s
493:	learn: 0.0971707	total: 7m 56s	remaining:

130:	learn: 0.1476735	total: 2m 4s	remaining: 5m 50s
131:	learn: 0.1473322	total: 2m 5s	remaining: 5m 49s
132:	learn: 0.1470089	total: 2m 6s	remaining: 5m 48s
133:	learn: 0.1466046	total: 2m 7s	remaining: 5m 47s
134:	learn: 0.1463539	total: 2m 8s	remaining: 5m 46s
135:	learn: 0.1461471	total: 2m 9s	remaining: 5m 45s
136:	learn: 0.1459479	total: 2m 10s	remaining: 5m 44s
137:	learn: 0.1454286	total: 2m 10s	remaining: 5m 43s
138:	learn: 0.1451646	total: 2m 11s	remaining: 5m 42s
139:	learn: 0.1448608	total: 2m 12s	remaining: 5m 41s
140:	learn: 0.1446701	total: 2m 13s	remaining: 5m 40s
141:	learn: 0.1444407	total: 2m 14s	remaining: 5m 39s
142:	learn: 0.1442836	total: 2m 15s	remaining: 5m 39s
143:	learn: 0.1438394	total: 2m 16s	remaining: 5m 38s
144:	learn: 0.1436246	total: 2m 17s	remaining: 5m 37s
145:	learn: 0.1432088	total: 2m 18s	remaining: 5m 36s
146:	learn: 0.1428059	total: 2m 19s	remaining: 5m 35s
147:	learn: 0.1423879	total: 2m 20s	remaining: 5m 34s
148:	learn: 0.1419924	total: 2m 21

283:	learn: 0.1166132	total: 4m 30s	remaining: 3m 25s
284:	learn: 0.1165412	total: 4m 31s	remaining: 3m 24s
285:	learn: 0.1164037	total: 4m 32s	remaining: 3m 23s
286:	learn: 0.1163237	total: 4m 33s	remaining: 3m 22s
287:	learn: 0.1162505	total: 4m 34s	remaining: 3m 21s
288:	learn: 0.1160974	total: 4m 35s	remaining: 3m 20s
289:	learn: 0.1160275	total: 4m 36s	remaining: 3m 19s
290:	learn: 0.1158719	total: 4m 37s	remaining: 3m 18s
291:	learn: 0.1156719	total: 4m 37s	remaining: 3m 17s
292:	learn: 0.1155719	total: 4m 38s	remaining: 3m 17s
293:	learn: 0.1154428	total: 4m 39s	remaining: 3m 16s
294:	learn: 0.1151722	total: 4m 40s	remaining: 3m 15s
295:	learn: 0.1150074	total: 4m 41s	remaining: 3m 14s
296:	learn: 0.1148269	total: 4m 42s	remaining: 3m 13s
297:	learn: 0.1147485	total: 4m 43s	remaining: 3m 12s
298:	learn: 0.1146046	total: 4m 44s	remaining: 3m 11s
299:	learn: 0.1145036	total: 4m 45s	remaining: 3m 10s
300:	learn: 0.1144307	total: 4m 46s	remaining: 3m 9s
301:	learn: 0.1142954	total: 

436:	learn: 0.1014136	total: 6m 56s	remaining: 1m
437:	learn: 0.1013693	total: 6m 57s	remaining: 59.1s
438:	learn: 0.1012189	total: 6m 58s	remaining: 58.2s
439:	learn: 0.1011601	total: 6m 59s	remaining: 57.2s
440:	learn: 0.1011157	total: 7m	remaining: 56.3s
441:	learn: 0.1010712	total: 7m 1s	remaining: 55.3s
442:	learn: 0.1010284	total: 7m 2s	remaining: 54.4s
443:	learn: 0.1009856	total: 7m 3s	remaining: 53.4s
444:	learn: 0.1009062	total: 7m 4s	remaining: 52.4s
445:	learn: 0.1008619	total: 7m 5s	remaining: 51.5s
446:	learn: 0.1006545	total: 7m 6s	remaining: 50.5s
447:	learn: 0.1006109	total: 7m 7s	remaining: 49.6s
448:	learn: 0.1004616	total: 7m 8s	remaining: 48.6s
449:	learn: 0.1003312	total: 7m 9s	remaining: 47.7s
450:	learn: 0.1002868	total: 7m 10s	remaining: 46.7s
451:	learn: 0.1002432	total: 7m 11s	remaining: 45.8s
452:	learn: 0.1001993	total: 7m 12s	remaining: 44.8s
453:	learn: 0.1001552	total: 7m 13s	remaining: 43.9s
454:	learn: 0.1000721	total: 7m 14s	remaining: 42.9s
455:	lear

93:	learn: 0.1313203	total: 1m 33s	remaining: 6m 42s
94:	learn: 0.1308697	total: 1m 34s	remaining: 6m 41s
95:	learn: 0.1306464	total: 1m 35s	remaining: 6m 39s
96:	learn: 0.1300919	total: 1m 36s	remaining: 6m 38s
97:	learn: 0.1298136	total: 1m 36s	remaining: 6m 37s
98:	learn: 0.1294001	total: 1m 37s	remaining: 6m 36s
99:	learn: 0.1290195	total: 1m 38s	remaining: 6m 35s
100:	learn: 0.1288107	total: 1m 39s	remaining: 6m 34s
101:	learn: 0.1280282	total: 1m 40s	remaining: 6m 33s
102:	learn: 0.1278160	total: 1m 41s	remaining: 6m 32s
103:	learn: 0.1276236	total: 1m 42s	remaining: 6m 31s
104:	learn: 0.1274351	total: 1m 43s	remaining: 6m 30s
105:	learn: 0.1269226	total: 1m 44s	remaining: 6m 30s
106:	learn: 0.1264300	total: 1m 46s	remaining: 6m 29s
107:	learn: 0.1260817	total: 1m 47s	remaining: 6m 29s
108:	learn: 0.1257471	total: 1m 48s	remaining: 6m 28s
109:	learn: 0.1255367	total: 1m 49s	remaining: 6m 27s
110:	learn: 0.1251768	total: 1m 50s	remaining: 6m 26s
111:	learn: 0.1249900	total: 1m 51s

246:	learn: 0.0963369	total: 4m 5s	remaining: 4m 11s
247:	learn: 0.0962496	total: 4m 6s	remaining: 4m 10s
248:	learn: 0.0961621	total: 4m 7s	remaining: 4m 9s
249:	learn: 0.0960747	total: 4m 8s	remaining: 4m 8s
250:	learn: 0.0958111	total: 4m 9s	remaining: 4m 7s
251:	learn: 0.0957240	total: 4m 10s	remaining: 4m 6s
252:	learn: 0.0956380	total: 4m 11s	remaining: 4m 5s
253:	learn: 0.0955523	total: 4m 12s	remaining: 4m 4s
254:	learn: 0.0953516	total: 4m 13s	remaining: 4m 3s
255:	learn: 0.0952672	total: 4m 14s	remaining: 4m 2s
256:	learn: 0.0951842	total: 4m 15s	remaining: 4m 1s
257:	learn: 0.0948210	total: 4m 16s	remaining: 4m
258:	learn: 0.0947288	total: 4m 17s	remaining: 4m
259:	learn: 0.0946438	total: 4m 18s	remaining: 3m 59s
260:	learn: 0.0945413	total: 4m 19s	remaining: 3m 58s
261:	learn: 0.0944569	total: 4m 20s	remaining: 3m 56s
262:	learn: 0.0943758	total: 4m 21s	remaining: 3m 55s
263:	learn: 0.0941592	total: 4m 22s	remaining: 3m 55s
264:	learn: 0.0940764	total: 4m 23s	remaining: 3m 

400:	learn: 0.0796795	total: 6m 43s	remaining: 1m 39s
401:	learn: 0.0794185	total: 6m 44s	remaining: 1m 38s
402:	learn: 0.0793618	total: 6m 44s	remaining: 1m 37s
403:	learn: 0.0793062	total: 6m 45s	remaining: 1m 36s
404:	learn: 0.0791732	total: 6m 46s	remaining: 1m 35s
405:	learn: 0.0789620	total: 6m 47s	remaining: 1m 34s
406:	learn: 0.0788794	total: 6m 48s	remaining: 1m 33s
407:	learn: 0.0788231	total: 6m 49s	remaining: 1m 32s
408:	learn: 0.0786182	total: 6m 50s	remaining: 1m 31s
409:	learn: 0.0785634	total: 6m 51s	remaining: 1m 30s
410:	learn: 0.0785076	total: 6m 52s	remaining: 1m 29s
411:	learn: 0.0784313	total: 6m 53s	remaining: 1m 28s
412:	learn: 0.0783755	total: 6m 54s	remaining: 1m 27s
413:	learn: 0.0781745	total: 6m 55s	remaining: 1m 26s
414:	learn: 0.0780826	total: 6m 56s	remaining: 1m 25s
415:	learn: 0.0780270	total: 6m 57s	remaining: 1m 24s
416:	learn: 0.0779717	total: 6m 58s	remaining: 1m 23s
417:	learn: 0.0779169	total: 6m 59s	remaining: 1m 22s
418:	learn: 0.0778624	total:

57:	learn: 0.1515202	total: 59.4s	remaining: 7m 32s
58:	learn: 0.1509954	total: 1m	remaining: 7m 31s
59:	learn: 0.1502886	total: 1m 1s	remaining: 7m 30s
60:	learn: 0.1493502	total: 1m 2s	remaining: 7m 29s
61:	learn: 0.1489277	total: 1m 3s	remaining: 7m 28s
62:	learn: 0.1483724	total: 1m 4s	remaining: 7m 27s
63:	learn: 0.1477179	total: 1m 5s	remaining: 7m 26s
64:	learn: 0.1473119	total: 1m 6s	remaining: 7m 26s
65:	learn: 0.1468047	total: 1m 7s	remaining: 7m 24s
66:	learn: 0.1462697	total: 1m 8s	remaining: 7m 23s
67:	learn: 0.1454860	total: 1m 9s	remaining: 7m 22s
68:	learn: 0.1450872	total: 1m 10s	remaining: 7m 20s
69:	learn: 0.1443101	total: 1m 11s	remaining: 7m 19s
70:	learn: 0.1436788	total: 1m 12s	remaining: 7m 18s
71:	learn: 0.1429405	total: 1m 13s	remaining: 7m 17s
72:	learn: 0.1425990	total: 1m 14s	remaining: 7m 15s
73:	learn: 0.1421921	total: 1m 15s	remaining: 7m 14s
74:	learn: 0.1418501	total: 1m 16s	remaining: 7m 13s
75:	learn: 0.1411888	total: 1m 17s	remaining: 7m 11s
76:	lea

211:	learn: 0.1019825	total: 3m 28s	remaining: 4m 43s
212:	learn: 0.1018834	total: 3m 29s	remaining: 4m 42s
213:	learn: 0.1017636	total: 3m 30s	remaining: 4m 41s
214:	learn: 0.1016688	total: 3m 31s	remaining: 4m 40s
215:	learn: 0.1015716	total: 3m 32s	remaining: 4m 39s
216:	learn: 0.1014775	total: 3m 33s	remaining: 4m 38s
217:	learn: 0.1012625	total: 3m 34s	remaining: 4m 37s
218:	learn: 0.1011185	total: 3m 35s	remaining: 4m 36s
219:	learn: 0.1010232	total: 3m 36s	remaining: 4m 35s
220:	learn: 0.1009296	total: 3m 37s	remaining: 4m 34s
221:	learn: 0.1006913	total: 3m 38s	remaining: 4m 33s
222:	learn: 0.1005968	total: 3m 39s	remaining: 4m 32s
223:	learn: 0.1004326	total: 3m 40s	remaining: 4m 31s
224:	learn: 0.1001634	total: 3m 41s	remaining: 4m 30s
225:	learn: 0.1000713	total: 3m 41s	remaining: 4m 29s
226:	learn: 0.0999009	total: 3m 42s	remaining: 4m 28s
227:	learn: 0.0996684	total: 3m 43s	remaining: 4m 27s
228:	learn: 0.0995748	total: 3m 44s	remaining: 4m 26s
229:	learn: 0.0994826	total:

364:	learn: 0.0838110	total: 6m	remaining: 2m 13s
365:	learn: 0.0835187	total: 6m 1s	remaining: 2m 12s
366:	learn: 0.0834356	total: 6m 2s	remaining: 2m 11s
367:	learn: 0.0832656	total: 6m 3s	remaining: 2m 10s
368:	learn: 0.0832027	total: 6m 4s	remaining: 2m 9s
369:	learn: 0.0831388	total: 6m 5s	remaining: 2m 8s
370:	learn: 0.0830776	total: 6m 6s	remaining: 2m 7s
371:	learn: 0.0829771	total: 6m 7s	remaining: 2m 6s
372:	learn: 0.0829142	total: 6m 7s	remaining: 2m 5s
373:	learn: 0.0828519	total: 6m 8s	remaining: 2m 4s
374:	learn: 0.0827897	total: 6m 10s	remaining: 2m 3s
375:	learn: 0.0827280	total: 6m 10s	remaining: 2m 2s
376:	learn: 0.0825606	total: 6m 11s	remaining: 2m 1s
377:	learn: 0.0825001	total: 6m 12s	remaining: 2m
378:	learn: 0.0822434	total: 6m 13s	remaining: 1m 59s
379:	learn: 0.0821599	total: 6m 14s	remaining: 1m 58s
380:	learn: 0.0819539	total: 6m 15s	remaining: 1m 57s
381:	learn: 0.0818933	total: 6m 16s	remaining: 1m 56s
382:	learn: 0.0818322	total: 6m 17s	remaining: 1m 55s


18:	learn: 0.1902659	total: 38.7s	remaining: 16m 19s
19:	learn: 0.1884579	total: 40.8s	remaining: 16m 18s
20:	learn: 0.1871432	total: 42.8s	remaining: 16m 16s
21:	learn: 0.1856703	total: 44.8s	remaining: 16m 14s
22:	learn: 0.1839542	total: 46.9s	remaining: 16m 12s
23:	learn: 0.1825499	total: 48.9s	remaining: 16m 9s
24:	learn: 0.1811147	total: 50.8s	remaining: 16m 5s
25:	learn: 0.1800847	total: 52.9s	remaining: 16m 3s
26:	learn: 0.1786290	total: 54.8s	remaining: 16m
27:	learn: 0.1774701	total: 56.9s	remaining: 15m 59s
28:	learn: 0.1763236	total: 59s	remaining: 15m 57s
29:	learn: 0.1751957	total: 1m 1s	remaining: 15m 56s
30:	learn: 0.1741154	total: 1m 3s	remaining: 15m 56s
31:	learn: 0.1732012	total: 1m 5s	remaining: 15m 57s
32:	learn: 0.1717874	total: 1m 7s	remaining: 15m 55s
33:	learn: 0.1710069	total: 1m 9s	remaining: 15m 53s
34:	learn: 0.1698742	total: 1m 11s	remaining: 15m 52s
35:	learn: 0.1687721	total: 1m 13s	remaining: 15m 49s
36:	learn: 0.1680535	total: 1m 15s	remaining: 15m 47s

170:	learn: 0.1172220	total: 5m 50s	remaining: 11m 15s
171:	learn: 0.1170080	total: 5m 52s	remaining: 11m 13s
172:	learn: 0.1168390	total: 5m 54s	remaining: 11m 10s
173:	learn: 0.1167421	total: 5m 56s	remaining: 11m 8s
174:	learn: 0.1165115	total: 5m 58s	remaining: 11m 6s
175:	learn: 0.1164212	total: 6m	remaining: 11m 4s
176:	learn: 0.1161459	total: 6m 2s	remaining: 11m 2s
177:	learn: 0.1159937	total: 6m 4s	remaining: 11m
178:	learn: 0.1158518	total: 6m 6s	remaining: 10m 58s
179:	learn: 0.1156226	total: 6m 8s	remaining: 10m 55s
180:	learn: 0.1154042	total: 6m 10s	remaining: 10m 53s
181:	learn: 0.1150879	total: 6m 12s	remaining: 10m 51s
182:	learn: 0.1148743	total: 6m 14s	remaining: 10m 49s
183:	learn: 0.1146803	total: 6m 16s	remaining: 10m 47s
184:	learn: 0.1144455	total: 6m 19s	remaining: 10m 45s
185:	learn: 0.1143560	total: 6m 21s	remaining: 10m 43s
186:	learn: 0.1141265	total: 6m 23s	remaining: 10m 41s
187:	learn: 0.1140309	total: 6m 25s	remaining: 10m 39s
188:	learn: 0.1139438	tota

322:	learn: 0.0972434	total: 11m 9s	remaining: 6m 6s
323:	learn: 0.0971937	total: 11m 11s	remaining: 6m 4s
324:	learn: 0.0971436	total: 11m 13s	remaining: 6m 2s
325:	learn: 0.0970938	total: 11m 15s	remaining: 6m
326:	learn: 0.0970437	total: 11m 17s	remaining: 5m 58s
327:	learn: 0.0969937	total: 11m 19s	remaining: 5m 56s
328:	learn: 0.0969443	total: 11m 21s	remaining: 5m 54s
329:	learn: 0.0968822	total: 11m 23s	remaining: 5m 52s
330:	learn: 0.0968340	total: 11m 25s	remaining: 5m 50s
331:	learn: 0.0967434	total: 11m 28s	remaining: 5m 48s
332:	learn: 0.0965111	total: 11m 30s	remaining: 5m 46s
333:	learn: 0.0964623	total: 11m 32s	remaining: 5m 44s
334:	learn: 0.0964042	total: 11m 34s	remaining: 5m 41s
335:	learn: 0.0963552	total: 11m 36s	remaining: 5m 39s
336:	learn: 0.0962734	total: 11m 38s	remaining: 5m 37s
337:	learn: 0.0961414	total: 11m 40s	remaining: 5m 35s
338:	learn: 0.0960860	total: 11m 42s	remaining: 5m 33s
339:	learn: 0.0959111	total: 11m 44s	remaining: 5m 31s
340:	learn: 0.0957

473:	learn: 0.0871628	total: 16m 19s	remaining: 53.7s
474:	learn: 0.0871248	total: 16m 21s	remaining: 51.7s
475:	learn: 0.0870865	total: 16m 23s	remaining: 49.6s
476:	learn: 0.0870483	total: 16m 25s	remaining: 47.5s
477:	learn: 0.0869108	total: 16m 27s	remaining: 45.5s
478:	learn: 0.0868726	total: 16m 30s	remaining: 43.4s
479:	learn: 0.0868334	total: 16m 32s	remaining: 41.3s
480:	learn: 0.0867951	total: 16m 34s	remaining: 39.3s
481:	learn: 0.0867569	total: 16m 36s	remaining: 37.2s
482:	learn: 0.0867193	total: 16m 38s	remaining: 35.1s
483:	learn: 0.0866820	total: 16m 40s	remaining: 33.1s
484:	learn: 0.0866444	total: 16m 42s	remaining: 31s
485:	learn: 0.0865581	total: 16m 44s	remaining: 28.9s
486:	learn: 0.0865207	total: 16m 46s	remaining: 26.9s
487:	learn: 0.0864833	total: 16m 48s	remaining: 24.8s
488:	learn: 0.0864458	total: 16m 50s	remaining: 22.7s
489:	learn: 0.0864090	total: 16m 52s	remaining: 20.7s
490:	learn: 0.0863719	total: 16m 54s	remaining: 18.6s
491:	learn: 0.0863352	total: 1

GridSearchCV(cv=2,
             estimator=<catboost.core.CatBoostClassifier object at 0x000001829B2E2820>,
             param_grid={'depth': [4, 6, 8], 'iterations': [100, 500],
                         'learning_rate': array([0.1, 0.2])},
             scoring=make_scorer(f1_score), verbose=3)

In [63]:
%%time
best_cat = CatBoostClassifier(**grid_cat.best_params_)
best_cat.fit(train_tfidf, y_train)

y_cat_pred = best_cat.predict(valid_tfidf)
f1_cat = f1_score(y_valid, y_cat_pred)

print("Best parameters: ", grid_cat.best_params_)
print("Best RMSE score: ", grid_cat.best_score_)

0:	learn: 0.5170536	total: 1.74s	remaining: 14m 26s
1:	learn: 0.4052554	total: 3.8s	remaining: 15m 46s
2:	learn: 0.3344588	total: 5.88s	remaining: 16m 14s
3:	learn: 0.2933226	total: 8.08s	remaining: 16m 41s
4:	learn: 0.2666289	total: 10.2s	remaining: 16m 45s
5:	learn: 0.2495567	total: 12.3s	remaining: 16m 56s
6:	learn: 0.2376379	total: 14.5s	remaining: 17m 1s
7:	learn: 0.2270916	total: 16.6s	remaining: 17m
8:	learn: 0.2207240	total: 18.7s	remaining: 17m
9:	learn: 0.2154455	total: 20.9s	remaining: 17m 2s
10:	learn: 0.2116262	total: 22.9s	remaining: 16m 59s
11:	learn: 0.2081980	total: 25.1s	remaining: 16m 58s
12:	learn: 0.2055707	total: 27.2s	remaining: 17m
13:	learn: 0.2026354	total: 29.3s	remaining: 16m 58s
14:	learn: 0.1993634	total: 31.4s	remaining: 16m 54s
15:	learn: 0.1970646	total: 33.4s	remaining: 16m 49s
16:	learn: 0.1953072	total: 35.4s	remaining: 16m 45s
17:	learn: 0.1928250	total: 37.4s	remaining: 16m 42s
18:	learn: 0.1902659	total: 39.5s	remaining: 16m 40s
19:	learn: 0.18845

153:	learn: 0.1205172	total: 5m 17s	remaining: 11m 52s
154:	learn: 0.1201636	total: 5m 19s	remaining: 11m 50s
155:	learn: 0.1200122	total: 5m 21s	remaining: 11m 48s
156:	learn: 0.1198309	total: 5m 23s	remaining: 11m 46s
157:	learn: 0.1195530	total: 5m 25s	remaining: 11m 44s
158:	learn: 0.1194488	total: 5m 27s	remaining: 11m 42s
159:	learn: 0.1191564	total: 5m 29s	remaining: 11m 39s
160:	learn: 0.1190370	total: 5m 31s	remaining: 11m 37s
161:	learn: 0.1186898	total: 5m 33s	remaining: 11m 35s
162:	learn: 0.1185905	total: 5m 35s	remaining: 11m 33s
163:	learn: 0.1183739	total: 5m 37s	remaining: 11m 31s
164:	learn: 0.1182557	total: 5m 39s	remaining: 11m 29s
165:	learn: 0.1181539	total: 5m 41s	remaining: 11m 26s
166:	learn: 0.1180071	total: 5m 43s	remaining: 11m 24s
167:	learn: 0.1178955	total: 5m 45s	remaining: 11m 22s
168:	learn: 0.1176327	total: 5m 47s	remaining: 11m 20s
169:	learn: 0.1173612	total: 5m 49s	remaining: 11m 18s
170:	learn: 0.1172220	total: 5m 51s	remaining: 11m 16s
171:	learn

305:	learn: 0.0984979	total: 10m 22s	remaining: 6m 34s
306:	learn: 0.0983881	total: 10m 24s	remaining: 6m 32s
307:	learn: 0.0983182	total: 10m 26s	remaining: 6m 30s
308:	learn: 0.0982672	total: 10m 28s	remaining: 6m 28s
309:	learn: 0.0982178	total: 10m 30s	remaining: 6m 26s
310:	learn: 0.0981680	total: 10m 32s	remaining: 6m 24s
311:	learn: 0.0981180	total: 10m 34s	remaining: 6m 22s
312:	learn: 0.0980672	total: 10m 36s	remaining: 6m 20s
313:	learn: 0.0979330	total: 10m 38s	remaining: 6m 18s
314:	learn: 0.0978271	total: 10m 40s	remaining: 6m 15s
315:	learn: 0.0977775	total: 10m 42s	remaining: 6m 13s
316:	learn: 0.0975411	total: 10m 44s	remaining: 6m 11s
317:	learn: 0.0974905	total: 10m 46s	remaining: 6m 9s
318:	learn: 0.0974404	total: 10m 48s	remaining: 6m 7s
319:	learn: 0.0973925	total: 10m 50s	remaining: 6m 5s
320:	learn: 0.0973433	total: 10m 51s	remaining: 6m 3s
321:	learn: 0.0972931	total: 10m 53s	remaining: 6m 1s
322:	learn: 0.0972434	total: 10m 55s	remaining: 5m 59s
323:	learn: 0.0

456:	learn: 0.0881644	total: 15m 24s	remaining: 1m 27s
457:	learn: 0.0880364	total: 15m 26s	remaining: 1m 25s
458:	learn: 0.0879986	total: 15m 28s	remaining: 1m 22s
459:	learn: 0.0879035	total: 15m 30s	remaining: 1m 20s
460:	learn: 0.0878649	total: 15m 32s	remaining: 1m 18s
461:	learn: 0.0878259	total: 15m 34s	remaining: 1m 16s
462:	learn: 0.0877877	total: 15m 36s	remaining: 1m 14s
463:	learn: 0.0877489	total: 15m 38s	remaining: 1m 12s
464:	learn: 0.0876609	total: 15m 40s	remaining: 1m 10s
465:	learn: 0.0875299	total: 15m 42s	remaining: 1m 8s
466:	learn: 0.0874909	total: 15m 44s	remaining: 1m 6s
467:	learn: 0.0874316	total: 15m 46s	remaining: 1m 4s
468:	learn: 0.0873932	total: 15m 48s	remaining: 1m 2s
469:	learn: 0.0873550	total: 15m 50s	remaining: 1m
470:	learn: 0.0872771	total: 15m 52s	remaining: 58.7s
471:	learn: 0.0872391	total: 15m 54s	remaining: 56.6s
472:	learn: 0.0872008	total: 15m 56s	remaining: 54.6s
473:	learn: 0.0871628	total: 15m 58s	remaining: 52.6s
474:	learn: 0.0871248	

In [64]:
x_features = vstack((train_tfidf, valid_tfidf))
y_features = pd.concat([y_train, y_valid], axis=0)

In [69]:
%%time
test_lgb = lgb.LGBMClassifier(**grid_lgb.best_params_)
test_lgb.fit(x_features, y_features)

y_test_pred = test_lgb.predict(test_tfidf)
f1_lgb = f1_score(y_test, y_test_pred)
print("Best f1 score на тестовой выборке: ", f1_test)

Best f1 score на тестовой выборке:  0.7801671003972058
Wall time: 32.2 s


In [66]:
model = DummyClassifier(strategy = 'constant', constant = 1)
model.fit(train_tfidf, y_train)
y_pred = model.predict(test_tfidf)
f1_dum = f1_score(y_test, y_pred)
print("F1 score: ", f1_dum)

F1 score:  0.18449965808069296


In [70]:
df_pred = pd.DataFrame({'predict': f1_lgb, 'true': y_test})
df_pred['predict'] = df_pred['predict'].astype(int)
df_pred.sample(15)

predict  true
29533         0     0
45806         0     0
94629         0     0
18711         0     0
42211         0     0
134621        0     0
41872         0     0
108871        0     0
59507         0     0
39790         0     0
32379         0     0
688           0     0
93529         0     0
103669        0     0
75245         0     0

Модель значительно лучше константной. CatBoost занимает кучу времени, поэтому побеждает lgb.

# Вывод
1. Данные загружены.
2. Текст лемматизирован.
3. Применен метод tf-idf.
4. Модели обучены на полученной матрице.
5. Модели градиентного бустинга прошли порог f1 меры в 0.75 и показали результат 0.78
6. Модель сильно лучше константной и готова к работе.